### PR TITLE
first solution for task2Lottery-OOP

### PR DESCRIPTION
**Pull request by ivankusur**

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public static String getRandomColor() {
+        int index = new Random().nextInt(Color.values().length);
+        return Color.values()[index].toString();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/ColorSupplier.java:8]: random should be a class field, with appropriate access modifier. Also, revisit the checklist and fix the line 8

File: src/main/java/core/basesyntax/Lottery.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        ball.setNumber(randomInt);
+        ball.setColor(ColorSupplier.getRandomColor());
+        return ball;
+    }
+}
+
+
+
+
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:19]: 5 lines must be removed

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class Lottery {
+    public static Ball getRandomBall() {
+        Ball ball = new Ball();
+        Random random = new Random();
+        int randomInt = random.nextInt(100);
+        ball.setNumber(randomInt);
+        ball.setColor(ColorSupplier.getRandomColor());
+        return ball;
+    }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:13]: as mentioned in common mistakes checklist, we should avoid static methods - remove static modifier from this method and from getRandomColor, make 100 a class constant, ColorSupplier and Random a class fields - this way we won't recreate them needlessly per each method invocation

File: src/main/java/core/basesyntax/Application.java [hunk 4]
```diff
@@ -1,7 +1,13 @@
 package core.basesyntax;
 
 public class Application {
+    private static final int NUMBER_OF_BALL = 3;
+
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        Ball[] balls = new Ball[NUMBER_OF_BALL];
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:7]: we don't need an array here

File: src/main/java/core/basesyntax/Application.java [hunk 5]
```diff
@@ -1,7 +1,13 @@
...
+        for (int i = 0; i < balls.length; i++) {
+            balls[i] = Lottery.getRandomBall();
+            System.out.println(balls[i].toString());
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:10]: toString will be called by default under the hood

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job on your first try!

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
 
 public class Application {
+    private static final int NUMBER_OF_BALL = 3;
+
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        Lottery lotteryObject = new Lottery();
+        for (int i = 0; i < NUMBER_OF_BALL; i++) {
+            Ball b = lotteryObject.getRandomBall();
+            System.out.println(b);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Application.java:10]:
```suggestion
        Lottery lottery = new Lottery();
        for (int i = 0; i < NUMBER_OF_BALL; i++) {
            Ball ball = lottery.getRandomBall();
            System.out.println(b);
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 7]
```diff
@@ -1,7 +1,18 @@
...
+
 public class ColorSupplier {
+    private static final int MAX_NUMBER = 100;
+    private final Random random = new Random();
+
     public String getRandomColor() {
-        return null;
+        Color[] colors = Color.values();
+        int index = random.nextInt(colors.length);
+        return colors[index].toString();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorSupplier.java:12]: https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery#use-name-for-getting-string-representation-of-enum-constants

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+public class Lottery extends ColorSupplier {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Lottery.java:3]:
```suggestion
public class Lottery  {
    private static final int MAX_NUMBER = 100;

      private final Random random = new Random();
      priate final ColorSupplier colorSupplier = new ColorSupplier();
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**